### PR TITLE
More robust smoke test script

### PR DESCRIPTION
**Pull request by dividedmind**
- Keep going on a failure and only error out at the end.
- Use registry v2 in docker-compose.yml.
- Use unique compose project names to prevent clobbering.

### CODE REVIEW

File: examples/docker-compose.yml [hunk 1]
```diff
@@ -1,16 +1,14 @@
 version: '2'
 services:
   conjur:
-    image: registry.tld/conjur-appliance-cuke-master:4.9-stable
+    image: registry2.itci.conjur.net/conjur-appliance-cuke-master:4.9-stable
```
Comment (dustinmm80) [inline examples/docker-compose.yml]: This should stay `registry.tld`, let's not hardcode in our internal endpoints. Jenkins is already set up that way, and it's just a line in `/etc/hosts` for us. 

https://github.com/conjurinc/docker-registry-ansible-aws#usage

File: examples/smoketest.sh [hunk 2]
```diff
@@ -8,11 +8,31 @@ OSES=(
...
+   COMPOSE_PROJECT_NAME=$COMPOSE_PROJECT_NAME-$BUILD_NUMBER
+fi
+
+export COMPOSE_PROJECT_NAME
+
+ALL_OK=1
+
 finish() {
-  if [ "$NOKILL" != "0" ]; then
+  if [ "$NOKILL" == "0" ]; then
```
Comment (dustinmm80) [inline examples/smoketest.sh]: good catch, thanks!

File: examples/smoketest.sh [hunk 3]
```diff
@@ -26,7 +46,7 @@ main() {
 
   for os in "${OSES[@]}"; do
     for i in $(seq 4); do
-      scenario$i $os
+      try scenario$i $os
```
Comment (dustinmm80) [inline examples/smoketest.sh]: does this act like we expect in jenkins? or might we get some green builds that are actually broken if we `try` without `catch`?

File: examples/docker-compose.yml (hunk 1 above)
Comment (dustinmm80) [inline examples/docker-compose.yml]: I added this as a commit to the PR.

File: examples/smoketest.sh (hunk 3 above)
Comment (dividedmind) [inline examples/smoketest.sh]: There is no `catch`. (Pun certainly intended.) `try` isn't bash, it's just a function I defined above.

File: examples/docker-compose.yml (hunk 1 above)
Comment (dividedmind) [inline examples/docker-compose.yml]: Right; I first worked from the assumption that wrong endpoint was being used. Thanks for fixing that! (BTW, I really wish we'd rename `registry.tld`.)

**Pull request merged**